### DISCUSSION TITLE
Example of retrieving data

### DISCUSSION THREAD

**Discussion by jwils0n** [player-ui/player#146] [Q&A] [answered]
Hey folks! I'm putting together a player-ui POC and it's been a _long time_ since I've been in the weeds. Going through the docs, I can't find a simple example of how to retrieve player data. I know there used to be a method on the exposed `player` (or something) global, but so much has changed 😭 .

**Comment by adierkens** [comment, marked as answer]
That's a good callout, we definitely need better docs around the `player` and `reactPlayer` objects. I added an issue for us to publish the API docs as well: https://github.com/player-ui/player/issues/147


The most common way to grab data from the player is get get ahold of the `state` object:

```tsx
player.getState()
reactPlayer.player.getState()
```

The methods change slightly if the flow you're interacting with is currently in-progress or not. If it's in-progress, then you'll use:

```tsx
state.controllers.data.get()
state.controllers.data.serialize()
```

or any of the other methods on the `DataController` 


If the flow you're using is completed, then you have access to a `data` property on the completed state, which is the serialized model. We'll likely also include the `controllers` on the `CompletedState` going forward in case you need more granular access to some of the sub-systems

**Reply by jwils0n** [reply]
You're the best @adierkens, that's exactly what I was hoping to find, thanks!
